### DISCUSSION TITLE
Error messages from pre-written code

### DISCUSSION THREAD

**Discussion by benpymer** [Part-II-Computational-Physics/.github#15] [Q&A] [answered]
Hi!

I keep getting an error message after the code that animates the explicit upwind scheme, but I don't understand what's going wrong. I've attached a screenshot of them here.

![Screenshot (28)](https://user-images.githubusercontent.com/125975405/220633693-2f803a93-8a93-4bcb-b6c5-5f0a75d1d716.png)

Some context, I had to change %matplotlib notebook to %matplotlib inline in the first line of code because VS Code doesn't like %matplotlib notebook for some reason, so it might have something to do with that. If this is the problem, is there an alternative %matplotlib widget that works in VS Code that will execute the commands correctly?

It also might have something to do with my code (I don't know Python very well as I intermitted last year and this part II cohort is the first cohort to do it from Part IA, we used to do Matlab and C++). As far as I can tell it runs fine, but here it is in case it's to do with that.

![Screenshot (29)](https://user-images.githubusercontent.com/125975405/220634686-4b40e368-2a58-4540-a6f3-f9ba09ac4fff.png)

Thanks! Ben :)

**Comment by AustenLamacraft** [comment, marked as answer]
Hi Ben,

You've accidentally indented the line `return step` so that it is now inside definition of the function `step`. As a result your function `Upwind_explicit` is not returning anything (`None`), which is why it can't be called by the line `step(f)`. 

By the way, it's clearer if you paste the contents of a cell rather than adding screenshots. Put the code in a markdown code block like this:

<pre>
```python
print("hello world")
```
</pre>

which will show up like this:


 ```python
print("hello world")
```

**Reply by benpymer** [reply]
Hi Austen,

Thank you very much, after doing that and removing the update of t I seem to have code that runs, although it produces a static picture rather than an animation. Is this what it's supposed to do?
